libs/openFrameworks/utils/ofFileUtils.cpp

```diff
@@ -1819,9 +1811,9 @@ string ofFilePath::getFileName(const of::filesystem::path& _filePath, bool bRela
 }
 
 //------------------------------------------------------------------------------------------------------------
-string ofFilePath::getBaseName(const of::filesystem::path& filePath){
-	// FIXME: is this still useful?
-	return ofFile(filePath,ofFile::Reference).getBaseName();
+// FIXME: - suggest using stem() instead
+string ofFilePath::getBaseName(const of::filesystem::path & filePath){
+	return filePath.stem().string();
```

Comment:
can we get a test that shows what happens with something like: 
```
myDir/myOtherDir
myDir/myOtherDir/
myDir/myOtherDir/myFile 
```

Reply:
I've just tested this code on master and this PR
```c++
	for (auto & s : {
		string{ "myDir/myOtherDir" },
		string{ "myDir/myOtherDir/" },
		string{ "myDir/myOtherDir/file" },
	}) {
		cout << ofFilePath::getBaseName(s) << endl;
	}
```

and the results are the same. I'm wondering if the second test is correct.
```
myOtherDir

file
```

Reply:
thanks for testing - at least they are the same!! 

not saying we should base our decisions on php 😁
but basename with a folder that has a / returns the folder name: 
https://www.php.net/manual/en/function.basename.php 

be curious if others had thoughts on this, but maybe that is a new issue/bugfix separate from this PR ?

Reply:
I've tested a variant with 		
```cout << of::filesystem::path(s).filename() << endl;```
and surprisingly the results are the same.

Reply:
to be clear `std::filesystem::path::stem` returns an empty string on a trailing /:
```
path: "/foo/bar.txt", stem: "bar"
path: "/foo/.bar", stem: ".bar"
path: "foo.bar.baz.tar", stem: "foo.bar.baz"
path: "foo/bar/", stem: ""
```
the OF `ofFilePath::getBaseName()` docs are ambiguous and does not mention the case of the trailing slash, so we can consider relying on it to represent a "directory" a form of undefined behaviour...

but as mentioned above most if not all usage of the `ofFilePath` stuff should be removed in favour of operating on std::filesystem::filepath instances (which probably covers all the functionality provided by static ofFilePath functions), and making sure the `filepath` nature of paths is preserved integrally up to the last minute (the function calls involved with the FS access itself). as mentioned elsewhere, both to reduce the problems linked to wstring, and also to stop reasoning about paths as strings.

so I guess this means for historical reasons we want `ofFilePath::getBaseName()` to behave like before (PHP-style or not) and we want to deprecate it and remove it from core.